chains/evm/calls/contracts/bridge/deposit.go

```diff
@@ -21,3 +22,46 @@ func ConstructPermissionlessGenericDepositData(metadata []byte, executionFunctio
 	data = append(data, metadata...)
 	return data
 }
+
+func constructMainDepositData(tokenStats *big.Int, destRecipient []byte) []byte {
+	var data []byte
+	data = append(data, math.PaddedBigBytes(tokenStats, 32)...)                            // Amount (ERC20) or Token Id (ERC721)
+	data = append(data, math.PaddedBigBytes(big.NewInt(int64(len(destRecipient))), 32)...) // length of recipient
+	data = append(data, destRecipient...)                                                  // Recipient
+	return data
+}
+
+func ConstructErc20DepositData(destRecipient []byte, amount *big.Int) []byte {
+	data := constructMainDepositData(amount, destRecipient)
+	return data
+}
+
+func ConstructErc20DepositDataWithPriority(destRecipient []byte, amount *big.Int, priority uint8) []byte {
```

Comment:
Everything with priority we don't need

Reply:
Almost feel like this entire file should rather sit in tests folder as that is where it is only used.

Reply:
Deleted those with priority. Except in tests, e.g. ConstructErc20DepositData is also used in Erc20Deposit method in BridgeContract, while BridgeContract is created in app.go. So leave it as it is for now?

Reply:
I would almost consider removing Erc20Deposit and just using the construct util functions in tests only as I don't really wanna keep test code with the production code if we don't have to. 
Feel free to decide I am fine either way for now.